api/api.go

```diff
@@ -66,3 +70,29 @@ func handleRotateCluster(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusAccepted)
 	outputJSON(c, w, cluster)
 }
+
+func handleDrainNode(c *Context, w http.ResponseWriter, r *http.Request) {
+
+	drainNodeRequest, err := model.NewDrainNodeRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to decode request")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	node := model.NodeDrain{
+		NodeName:                drainNodeRequest.NodeName,
+		GracePeriod:             drainNodeRequest.GracePeriod,
+		MaxDrainRetries:         drainNodeRequest.MaxDrainRetries,
+		WaitBetweenPodEvictions: drainNodeRequest.WaitBetweenPodEvictions,
+		DetachNode:              drainNodeRequest.DetachNode,
+		TerminateNode:           drainNodeRequest.TerminateNode,
+		ClusterID:               drainNodeRequest.ClusterID,
+	}
+
+	go rotator.InitDrainNode(&node, c.Logger.WithField("node", node.NodeName))
```

Comment:
this doesn't guarantee that it won't fail. There is a library where you can handle the errors https://github.com/oklog/run

Reply:
it's ok to skip for now, just keep this in mind